curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -932,45 +932,78 @@ void FileCacheManager::TruncateCache(uint64_t offset, uint64_t fileSize) {
 }
 
 CURVEFS_ERROR FileCacheManager::Flush(bool force, bool toS3) {
-    CURVEFS_ERROR ret;
+    CURVEFS_ERROR ret = CURVEFS_ERROR::OK;
     std::map<uint64_t, ChunkCacheManagerPtr> tmp;
     {
         WriteLockGuard writeLockGuard(rwLock_);
         tmp = chunkCacheMap_;
     }
-    auto iter = tmp.begin();
-
-    for (; iter != tmp.end(); iter++) {
-        ret = iter->second->Flush(inode_, force, toS3);
-        if (ret != CURVEFS_ERROR::OK) {
-            LOG(ERROR) << "fileCacheManager Flush error, ret:" << ret
-                       << ",chunkIndex:" << iter->second->GetIndex();
-            return ret;
-        }
 
-        {
-            WriteLockGuard writeLockGuard(rwLock_);
-            auto iter1 = chunkCacheMap_.find(iter->first);
-            if (iter1 == chunkCacheMap_.end()) {
-                VLOG(1) << "Flush, chunk cache for inodeid: " << inode_
+    std::atomic<uint64_t> pendingReq(0);
+    curve::common::CountDownEvent cond(1);
+    FlushChunkCacheCallBack cb =
+        [&](const std::shared_ptr<FlushChunkCacheContext> &context) {
+            ret = context->retCode;
```

Comment:
The context is shared by multiple threads, is there no need to lock here?

Reply:
fix